recipe/build.sh

```diff
@@ -1,5 +1,11 @@
 #! /bin/bash
 
-meson --prefix=$PREFIX --libdir=lib -Dexamples=false -Dutils=false build
+meson \
+  --prefix="${PREFIX}" \
+  --libdir=lib \
+  --sbindir=bin \
```

Comment:
I wonder what's the best thing to do with this.
With `--sbindir=bin` we get `fusermount3` at `$PREFIX/bin` and hence on `$PATH` -- which is what we want so we have `fusermount3` readily available.
OTOH, the `fusermount3` you'll get from the Conda package won't have the setuid bit set. Which means if a user has `fusermount3` already installed from its Linux distribution and relies on it being setuid, then our non-setuid `fusermount3` won't work for them (unless they invoke their system's one with an absolute path).

I think I'm more inclined to say let's go with `$PREFIX/bin` (i.e., the state of this PR) and let the user handle setuid/absolute path invocation. Otherwise we'd probably have to patch downstream packages to use `$PREFIX/sbin` if we want ours or we need the user to put `$PREFIX/sbin` on `$PATH` by themselves -- which seems more hassle than worth it, IMHO.

Reply:
How common is it for `fusermount3` to require setuid? Is it a just a case of old Linux kernels not supporting the purely userspace version?

There is some precedent for requiring users to use absolute path invocation for `kinit` (e.g. macOS keychain integration, some non-standard kerberos setups) so I'm inclined to agree with you regardless.

Reply:
> How common is it for `fusermount3` to require setuid? Is it a just a case of old Linux kernels not supporting the purely userspace version?

I just checked by installing `fuse3` in containers for `centos:7`, `debian:10-slim`, `ubuntu:20.04`, `fedora:33` and all of them have `fusermount3` with setuid -- which makes sense because it's meant to be invoked by users but in the end requires privileges for the mount.
https://github.com/libfuse/libfuse/tree/fuse-3.10.0#security-implications also explicitly says "The *fusermount3* program is installed setuid root.", so it would be odd if the Linux distributions didn't do it :).

> There is some precedent for requiring users to use absolute path invocation for `kinit` (e.g. macOS keychain integration, some non-standard kerberos setups) so I'm inclined to agree with you regardless.

Thanks, that's all I wanted to head :).

Reply:
Hmm, if `fusermount3` is completely non-functional without having setuid then I'm inclined to disagree with myself :)

A middle ground could be to add a small wrapper in `$CONDA_PREFIX/bin` that:
1. If `PREFIX/sbin/fusermount3` has setuid set, use it
2. If `fusermount3` is on PATH use that instead
3. Show an error explaining what's wrong

Any thoughts?

Reply:
> I'm inclined to disagree with myself :)

^_^
Well, what can I say, I'm even one of those people that has (more often than one would like) multiple (often conflicting XD) opinions one a single topic -- at the _same time_, that is XD.

> A middle ground could be to add a small wrapper in `$CONDA_PREFIX/bin` ...

I think maybe we should keep this more self-contained (i.e., not exec'ing `/sbin/fusermount3` only in some situations) to avoid any inconsistent behavior users may observe. Also, our `fusermount3` never has setuid set, so we needn't test for that.
Hm...
I'd be fine by me if we go for the conservative road then and use `PREFIX/sbin` for now. Then people/downstream packages can decide on their own -- and we are free to add a `PREFIX/bin/fusermount3` wrapper in a later build whenever there is request for it.
WDYT?

Reply:
Yeah, I agree with your conclusion.

I definitely think we shouldn't have an always-broken `fusermount3` on `$PATH` (well unless you're running as root). My thought with the wrapper was that it could tell people to add the setuid bit if there isn't a suitable installation on the underlying OS. Though it's something which could easily be added later if it turns out to be necessary for making this package usable.